delegated_translator.go

```diff
@@ -121,11 +121,30 @@ func (dt *delegatedTranslator) find(w http.ResponseWriter, r *http.Request) {
 	res := parsed.MultihashResults[0]
 
 	out := drResp{}
+
+	// Records returned from IPNI via Delegated Routing don't have ContextID in them. Becuase of that,
+	// some records that are valid from the IPNI point of view might look like duplicates from the Delegated Routing point of view.
+	// To make the Delegated Routing output nicer, deduplicate identical records.
+	uniqueProviders := map[string]struct{}{}
+	appendIfUnique := func(drp *drProvider) {
+		drpb := make([]byte, 0, len(drp.ID)+len(drp.Protocol)+len(drp.Schema)+len(drp.Metadata))
+		drpb = append(drpb, []byte(drp.ID)...)
+		drpb = append(drpb, []byte(drp.Protocol)...)
+		drpb = append(drpb, []byte(drp.Schema)...)
+		drpb = append(drpb, drp.Metadata...)
+		drps := string(drpb)
```

Comment:
can we have this be a `String()` method on `drProvider`?

Reply:
I thought about that too initially. The reason I've done it not in the `String` method - is because this is a specific string representation that exist for one specific purpose (deduplication). I.e. it's not a generic string representation of the object (which to me `String` is). But I've been lacking a second opinion. Happy to move it to `drProvider.String` if you feel that it's going to fit it better.

Reply:
Should we hash this value to an int via crc32 or something similar. As is the memory footprint seems large for what it's doing.

As an example, see https://github.com/ipni/indexstar/blob/main/find_ndjson.go#L44

Reply:
Fair point - fixed